libobs/util/platform-cocoa.m

```diff
@@ -505,3 +506,94 @@ bool cfstr_copy_dstr(CFStringRef cfstring, CFStringEncoding cfstring_encoding, s
 
     return (bool) success;
 }
+
+bool os_keychain_save(const char *key, const char *data)
+{
+    if (!key || !data)
+        return false;
+
+    NSString *nsKey = [[NSString alloc] initWithBytesNoCopy:(void *) key length:strlen(key)
+                                                   encoding:NSUTF8StringEncoding
+                                               freeWhenDone:NO];
+    NSData *nsData = [NSData dataWithBytesNoCopy:(void *) data length:strlen(data) freeWhenDone:NO];
```

Comment:
As you ensure that `key` is not an invalid pointer, you should be able to just box the char pointer to an `NSString` via `NSString *nsKey = @(key);`.

Alternatively you could just use CoreFoundation types throughout:

```
CFStringRef account = CFStringCreateWithCString(kCFAllocatorDefault, key, kCFStringEncodingUTF8);
CFStringRef service = CFStringCreateWithCString(kCFAllocatorDefault, "OBS Studio", kCFStringEncodingUTF8);

CFDataRef = CFDataCreate(kCFAllocatorDefault, data, strlen(data));

CFTypeRef keys[4] = {kSecAttrAccessible, kSecClass, kSecAttrService,
                 kSecAttrAccount, kSecValueData};

CFTypeRef values[4] = {kSecAttrAccessibleWhenUnlocked, kSecClassGenericPassword,
                   service, account, data};
```

It's functionally probably not much different, but personally I prefer to use CoreFoundation types when interacting in the CoreFoundation world.

You might need to `CFRelease` some of them after you're done of course.

Reply:
I assume I could use `CFStringCreateWithCStringNoCopy`/`CFDataCreateWithBytesNoCopy` to avoid copying (since that not necessary here) in the same way as its done with `NSString`/`NSData`.
I'd still prefer to avoid using a CFDictionary directly since the readability suffers a lot.

Reply:
Fair enough - we use `CFDictionary` in `obs-outputs` already to access to root certificates in the keychain, so it would be "familiar". And the readability improves the longer you work with CoreFoundation. 😉